torch/_utils.py

```diff
@@ -463,8 +463,17 @@ def _get_devices_properties(device_ids):
     # all device properties
     return [_get_device_attr(lambda m: m.get_device_properties(i)) for i in device_ids]
 
+def get_current_device_index() -> int:
```

Comment:
Underscore me?

Reply:
The eager version of the API has an `_` -> `_get_current_device_index`. So couldn't use. Do you suggest changing the name instead?

Reply:
Can you unify eager and jit implementations for get_current_device_index? It looks like they are doing the same thing, spelled slightly differently, like, using `torch.cuda.is_available()` vs `torch.cuda.device_count()`

Reply:
`_get_current_device_index` currently uses `lambda` expression which is not supported in JIT. That's the reason for introducing a separate API for JIT

Reply:
The implementation is pretty simple:

```
def _get_device_attr(get_member):
    device_type = _get_available_device_type()
    if device_type and device_type.lower() == "cuda":
        return get_member(torch.cuda)
    # add more available device types here
    return None


def _get_current_device_index():
    # current device index
    return _get_device_attr(lambda m: m.current_device())
```

TBH, you should just inline the lambda and trace it directly

Reply:
If perf of eager and jit implementation is the same, then eager can use your jit implementation. You can use instruction counts to measure perf (`collect_callgrind` https://pytorch.org/tutorials/recipes/recipes/benchmark.html), timer may be too noisy.